Ai 303/pricing multiple images t2i i2i

pschroedl opened the pull request:
**What does this pull request do? Explain your changes. (required)**
The num_images_per_prompt field in text-to-image and image-to-image requests is multiplied by the calculated pixels of a requested image to produce a price proportionate to the request.

**Specific updates (required)**
* adds check for req.NumImagesPerPrompt and is multiplied by outPixels if > 1 ( default )
* removes numImage from latency calculations, as this is already factored into outPixels/
* update AIRequestFinished to factor numImages into pricePerAIUnit
* account for numImages in orchestrator debit fee calculation and add redundant nil check for backward-compatibility

**How did you test each of these updates (required)**

Manual Testing t2i:
aiModels.json
```
[
    {
        "pipeline": "text-to-image",
        "model_id": "ByteDance/SDXL-Lightning",
        "price_per_unit": 5000001,
        "warm": true
    }
]
```
numImages = 1
```
curl -X POST "http://0.0.0.0:8935/text-to-image"     -H "Content-Type: application/json"     -d '{
        "model_id":"ByteDance/SDXL-Lightning",
        "prompt":"A cool cat on the beach",
        "width": 1024,
        "height": 1024,
        "num_images_per_prompt": 1
    }'
```

```I0811 23:46:34.468360  237513 ai_http.go:280] manifestID=27_ByteDance/SDXL-Lightning orchSessionID=dc9cc8a0 clientIP=127.0.0.1 Received request id=bbc26d04 cap=27 modelID=ByteDance/SDXL-Lightning
I0811 23:46:34.471482  237513 orchestrator.go:246] manifestID=27_ByteDance/SDXL-Lightning orchSessionID=dc9cc8a0 clientIP=127.0.0.1 Payment tickets processed sessionID=27_ByteDance/SDXL-Lightning faceValue=600000000000000 WEI winProb=0.0200000000 ev=5999999999998.00
I0811 23:46:37.177075  237513 ai_http.go:321] manifestID=27_ByteDance/SDXL-Lightning orchSessionID=dc9cc8a0 clientIP=127.0.0.1 Processed request id=bbc26d04 cap=27 modelID=ByteDance/SDXL-Lightning took=2.705382698s
```

faceValue=600000000000000 WEI winProb=0.0200000000 ev=5999999999998.00

----

numImages = 3

```
curl -X POST "http://0.0.0.0:8935/text-to-image"     -H "Content-Type: application/json"     -d '{
        "model_id":"ByteDance/SDXL-Lightning",
        "prompt":"A cool cat on the beach",
        "width": 1024,
        "height": 1024,
        "num_images_per_prompt": 3
    }'
```

```
I0811 23:47:24.099706  237513 ai_http.go:280] manifestID=27_ByteDance/SDXL-Lightning orchSessionID=dc9cc8a0 clientIP=127.0.0.1 Received request id=ca8151a6 cap=27 modelID=ByteDance/SDXL-Lightning
I0811 23:47:24.100275  237513 orchestrator.go:468] Setting fixed price=130000026/25 for session=27_ByteDance/SDXL-Lightning
I0811 23:47:24.107618  237513 orchestrator.go:246] manifestID=27_ByteDance/SDXL-Lightning orchSessionID=dc9cc8a0 clientIP=127.0.0.1 Payment tickets processed sessionID=27_ByteDance/SDXL-Lightning faceValue=1800000000000000 WEI winProb=0.0600000000 ev=17999999999994.00
I0811 23:47:28.543152  237513 block_watcher.go:454] Polling blocks from=241909729 to=241909746
I0811 23:47:30.259654  237513 ai_http.go:321] manifestID=27_ByteDance/SDXL-Lightning orchSessionID=dc9cc8a0 clientIP=127.0.0.1 Processed request id=ca8151a6 cap=27 modelID=ByteDance/SDXL-Lightning took=6.151849654s
```
faceValue=1800000000000000 WEI winProb=0.0600000000 ev=17999999999994.00

Note the first response faceValue, winProb, and ev in the second response are the expected 3x values.

**Does this pull request close any open issues?**
Closes AI-303


**Checklist:**
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ x] Read the [contribution guide](./CONTRIBUTING.md)
- [ x] `make` runs successfully
- [ ] All tests in `./test.sh` pass ( not passing on ai-video )
- [ x] README and other documentation updated ( updated relevant comments )
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

pschroedl committed:
use NumImagesPerPrompt in t2i,i2i pricing calc

pschroedl committed:
update latency score calculations

rickstaa committed:
fix: account for numImages in orchestrator debit fee calculation

This commit updates the orchestrator to consider the number of images when
calculating DebitFees. It also corrects the logic for the pricePerAIUnit
metric and ensures the gateway sets a default number of images when not
provided, preventing nil errors in the orchestrator.

rickstaa reviewed (changes requested):
@pschroedl, thank you for your contribution! 🚀

I reviewed your pull request, and everything appears to be working well on the Gateway side. However, I had to make some minor changes to ensure the logic also works on the orchestrator side. You can see the changes here: [commit link](https://github.com/livepeer/go-livepeer/pull/3126/commits/b9a8134ad2983cca95ea226ffd83cee38b3afd54). I apologize for not mentioning this earlier in the [Linear issue](https://linear.app/livepeer-ai/issue/AI-303/ensure-t2ii2i-pricing-takes-number-of-images-into-account). Could you quickly check that my logic works out.

Additionally, I noticed that line 175 in [server/ai_process.go](https://github.com/livepeer/go-livepeer/blob/b1a611cd36ce54b8fb0366c017db3341be4990c8/server/ai_process.go#L175) should not be there. This line shows the price per unit, not the total price of a request. Although we don't have documentation for this yet, you can refer to the following metric pull requests for more context: [PR 3097](https://github.com/livepeer/go-livepeer/pull/3097) and [PR 3087](https://github.com/livepeer/go-livepeer/pull/3087).

rickstaa commented on server/ai_http.go:
```diff
@@ -196,7 +196,7 @@ func handleAIRequest(ctx context.Context, w http.ResponseWriter, r *http.Request
 		}
 
-		outPixels = height * width
+		outPixels = height * width * int64(*v.NumImagesPerPrompt)
```
@pschroedl we could add a `nil` check here aswell but I removed it since it should not occur with the code I now added at the gateway.

pschroedl replied:
I like the early check and reassignment to default if nil, particulary understanding that ultimately we will want middleware to perform the same.

rickstaa replied:
As discussed would be nice to have a nil check here as well just it ensure backward compatibility.

pschroedl replied:
Added.

rickstaa commented on server/ai_http.go:
```diff
@@ -215,7 +215,8 @@ func handleAIRequest(ctx context.Context, w http.ResponseWriter, r *http.Request
 		}
-		outPixels = int64(config.Height) * int64(config.Width)
+
+		outPixels = int64(config.Height) * int64(config.Width) * int64(*v.NumImagesPerPrompt)
```
Idem.

pschroedl replied:
Added.

rickstaa committed:
refactor(ai): prevent nil errors for NumImagesPerPrompt in orchestrator

This commit ensures the orchestrator handles cases where NumImagesPerPrompt
is not set in the request to prevent nil errors. Although the gateway enforces
this value, this change supports backward compatibility.

rickstaa reviewed (approved):
Looks great to me thanks 🙏🏻!
